src/core/bond_breakage/actions.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2022 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CORE_BOND_BREAKAGE_ACTIONS_HPP
+#define CORE_BOND_BREAKAGE_ACTIONS_HPP
+
+#include <boost/functional/hash.hpp>
+#include <boost/variant.hpp>
+
+#include <cstddef>
+
+namespace BondBreakage {
+
+// Delete Actions
+struct DeleteBond {
+  int particle_id;
+  int bond_partner_id;
+  int bond_type;
+  std::size_t hash_value() const {
+    std::size_t seed = 3875;
```

Comment:
How do the seed magic values come about?

Reply:
There is a discussion in [walberla/walberla#160](https://i10git.cs.fau.de/walberla/walberla/-/issues/160) (this actually where this code comes from). Some people like to use the golden ratio sequence because of its entropy, but the choice is somewhat arbitrary. Since the number of bonds is bounded to a small value, probably to the order of 100 for very large simulations, entropy is not as much an issue as it was for walberla. The seed differ between the two structs to avoid collisions, since both can appear in the same `std::vector`.